compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -403,12 +406,35 @@ public void visit(BLangResourceFunction funcNode, AnalyzerData data) {
             BLangType restParamType = funcNode.resourcePathType.restParamType;
             symResolver.resolveTypeNode(restParamType, data.env);
             if (!types.isAssignable(restParamType.getBType(), symTable.pathParamAllowedType)) {
-                dlog.error(restParamType.getPosition(), DiagnosticErrorCode.UNSUPPORTED_REST_PATH_PARAM_TYPE, 
+                dlog.error(restParamType.getPosition(), DiagnosticErrorCode.UNSUPPORTED_REST_PATH_PARAM_TYPE,
                         restParamType.getBType());
             }
         }
     }
 
+    private void checkForClientObjectTypeOrFunctionType(BType type, BType referredType, Location pos) {
+        if (referredType.tsymbol != null && Symbols.isFlagOn(referredType.tsymbol.flags, Flags.CLIENT)) {
```

Comment:
Why do we need this(`referredType.tsymbol != null`) null check?

Reply:
There are some `other` type have a `null` vale for `type.tsymbol`.

Reply:
So if the type is a semantic error we can return

Reply:
I will check it

Reply:
Updated 2725ba33d9230d4c98b3b79818080c3ba31e21b3